client/app/src/main/java/com/microsoft/research/karya/ui/registration/OTPFragment.kt

```diff
@@ -1,34 +1,110 @@
 package com.microsoft.research.karya.ui.registration
 
+import android.content.Context
 import android.os.Bundle
 import android.text.Editable
 import android.text.TextWatcher
 import androidx.fragment.app.Fragment
 import android.view.View
-import androidx.lifecycle.lifecycleScope
+import androidx.fragment.app.activityViewModels
+import androidx.lifecycle.Observer
 import androidx.navigation.fragment.findNavController
-import com.google.gson.JsonObject
 import com.microsoft.research.karya.R
+import com.microsoft.research.karya.data.model.karya.enums.OtpSendState
+import com.microsoft.research.karya.data.model.karya.enums.OtpVerifyState
 import com.microsoft.research.karya.data.service.KaryaAPIService
-import com.microsoft.research.karya.databinding.FragmentOTPBinding
+import com.microsoft.research.karya.databinding.FragmentOtpBinding
 import com.microsoft.research.karya.ui.base.BaseActivity
 import com.microsoft.research.karya.utils.viewBinding
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.launch
+import dagger.hilt.android.AndroidEntryPoint
 
 private const val OTP_LENGTH = 6
 
-class OTPFragment : Fragment(R.layout.fragment_o_t_p) {
+@AndroidEntryPoint
+class OTPFragment : Fragment(R.layout.fragment_otp) {
 
-    private val binding by viewBinding(FragmentOTPBinding::bind)
+    private val binding by viewBinding(FragmentOtpBinding::bind)
+    private val viewModel by activityViewModels<RegistrationViewModel>()
 
     private lateinit var registrationActivity: RegistrationActivity
     private lateinit var baseActivity: BaseActivity
     private lateinit var karyaAPI: KaryaAPIService
 
+    private fun setupObservers() {
+        viewModel.openDashBoardFromOTP.observe(viewLifecycleOwner, Observer { openDashBoard ->
+            if (openDashBoard) {
+                navigateToDashBoard()
+            }
+        })
+
+        viewModel.openProfilePictureFragmentFromOTP.observe(viewLifecycleOwner, { openProfilePictureFragment ->
+            if (openProfilePictureFragment) {
+                navigateToProfilePicture()
+            }
+        })
```

Comment:
Inconsistent usage of observer with and without explicit `Observer` class. Prefer using a plain lambda everywhere.

Reply:
Also move the lambda out of the parentheses